api/api_ethereum.go

```diff
@@ -1113,7 +1113,9 @@ func (args *EthTransactionArgs) setDefaults(ctx context.Context, b Backend) erro
 				}
 				args.MaxFeePerGas = (*hexutil.Big)(gasFeeCap)
 			}
-			if !isMagma && (args.MaxPriorityFeePerGas.ToInt().Cmp(gasPrice) != 0 || args.MaxFeePerGas.ToInt().Cmp(gasPrice) != 0) {
+			if isMagma && args.MaxFeePerGas.ToInt().Cmp(new(big.Int).Div(gasPrice, common.Big2)) < 0 {
```

Comment:
I think you should separate the 'isMagma' check. 
Otherwise still the 'else if' check logic can be operated when it's a magma hard fork.